mpi-pytest fails with firedrakeproject/firedrake:latest docker image

The following MFE 

```python
import pytest

@pytest.mark.parallel(nprocs=5)  # run in parallel with 5 processes
def test_my_code_on_5_procs():
    print("hello")
```

fails with


```bash
item = <Function test_my_code_on_5_procs>

    @pytest.hookimpl()
    def pytest_runtest_setup(item):
        from mpi4py import MPI

        if not _plugin_in_use:
            return

        if item.get_closest_marker("parallel"):
            if MPI.COMM_WORLD.size == 1:
                # If using pytest-mpi in "forking" mode, add a callback to item
                # that calls mpiexec
                assert not _is_parallel_child_process()
>               _set_parallel_callback(item)

usr/local/lib/python3.12/dist-packages/pytest_mpi/plugin.py:123:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
usr/local/lib/python3.12/dist-packages/pytest_mpi/plugin.py:234: in _set_parallel_callback
    impl = detect_mpi_implementation()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    def detect_mpi_implementation() -> MPIImplementation:
        try:
            result = subprocess.run(
                ["mpiexec", "--version"],
                stdout=subprocess.PIPE,
                stderr=subprocess.STDOUT,
                text=True,
                check=True
            )
        except FileNotFoundError:
            raise FileNotFoundError(
                "'mpiexec' not found on your PATH, please run in non-forking mode "
                "where you can specify a different MPI executable"
            )

        output = result.stdout.lower()
        if "open mpi" in output or "open-rte" in output:
            return MPIImplementation.OPENMPI
        elif "mpich" in output:
            return MPIImplementation.MPICH
        else:
>           raise RuntimeError(
                "MPI distribution is not recognised, please run in non-forking "
                "mode where you can specify your MPI executable"
            )
E           RuntimeError: MPI distribution is not recognised, please run in non-forking mode where you can specify your MPI executable

usr/local/lib/python3.12/dist-packages/pytest_mpi/plugin.py:323: RuntimeError
============================================================================================================================================================== short test summary info ===============================================================================================================================================================
ERROR test_mfe.py::test_my_code_on_5_procs - RuntimeError: MPI distribution is not recognised, please run in non-forking mode where you can specify your MPI executable
```

when one uses firedrake's latest docker image.

The reason is that an old version of openmpi is installed in that docker image

```
root@ec23963d2ad5:/# mpiexec --version
mpiexec (OpenRTE) 4.1.6

Report bugs to http://www.open-mpi.org/community/help/
```

which is not recognized by mpi-pytest in the conditionals

```python
        if "open mpi" in output or "open-rte" in output:
            return MPIImplementation.OPENMPI
        elif "mpich" in output:
            return MPIImplementation.MPICH
```

--- a/pytest_mpi/plugin.py
+++ b/pytest_mpi/plugin.py
@@ -315,7 +315,9 @@ def detect_mpi_implementation() -> MPIImplementation:
             "where you can specify a different MPI executable"
         )
 
-    if any(
+    if "mpich" in version:
+        return MPIImplementation.MPICH
+    elif any(
         version_str in version
         for version_str in [
             "open mpi",
@@ -327,8 +329,6 @@ def detect_mpi_implementation() -> MPIImplementation:
         ]
     ):
         return MPIImplementation.OPENMPI
-    elif "mpich" in version:
-        return MPIImplementation.MPICH
     elif "microsoft" in version:
         return MPIImplementation.MSMPI
     else:

